### ISSUE TITLE
Some networks not working with Imagenet in python

### ISSUE THREAD

**Issue by hoaphumanoid** [dmlc/mxnet#3359]
I've been experimenting with [Imagenet](https://github.com/dmlc/mxnet/blob/master/example/image-classification/train_imagenet.py) example in python. Inception-bn, inception-bn-full, alexnet, vgg networks works well. 

However, if I try to do the same with googlenet I get the following error:
```python
mxnet.base.MXNetError: InferShape Error in pooling4: [10:13:58] src/operator/./pooling-inl.h:198: Check failed: param_.kernel[0] <= dshape[2] + 2 * param_.pad[0] && param_.kernel[1] <= dshape[3] + 2 * param_.pad[1] kernel size exceed input
```
With resnet (after chaning the ptyhon code so if it accepted):
```python
File "/usr/local/lib/python2.7/dist-packages/mxnet-0.7.0-py2.7.egg/mxnet/base.py", line 77, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [10:09:43] src/storage/./pooled_storage_manager.h:67: Memory allocation failed.
```
With inception-v3:
```python
mxnet.base.MXNetError: InferShape Error in global_pool: [10:22:42] src/operator/./pooling-inl.h:198: Check failed: param_.kernel[0] <= dshape[2] + 2 * param_.pad[0] && param_.kernel[1] <= dshape[3] + 2 * param_.pad[1] kernel size exceed input
```

**Comment by piiswrong** [comment]
Your input is too small/kernels are too big

**Comment by hoaphumanoid** [comment]
But the input is exactly the same as in the other networks. Are you sure that the code in the repo is working?

**Comment by mli** [comment]
inception-v3: the input size should be >= 299*299
resnet: reduce the batch size

**Comment by hoaphumanoid** [comment]
I reduced the batch size of resnet to 72. I don't get OOM but the network is not training, I found that it is meant for CIFAR images of [32x32](https://github.com/dmlc/mxnet/blob/master/example/image-classification/symbol_resnet.py#L3), not for Imagenet (224x224)

Regarding inception-v3, I used a data set of 480 and then put `--data-shape 299`, and everything worked perfectly. 

Thanks!

**Comment by hoaphumanoid** [comment]
I just saw that alexnet and vgg were not working for me (the others are):

`python train_imagenet.py --batch-size 144 --lr 0.05 --lr-factor .94 --gpus 0,1,2,3 --num-epoch 60 --network alexnet`
Not converging, accuracy: 0.000833, 0.000694, 0.000833, 0.001111, 0.000278...
The same with shape 299 and initial images of size 480 and quality 90:
Not converging, accuracy: 0.000833, 0.000694, 0.000972, 0.000556, 0.000972, 0.001111, 0.001250, 0.001250, 0.001667, 0.000278, 0.000694, 0.000972, 0.001389, 0.000694

`python train_imagenet.py --batch-size 144 --lr 0.05 --lr-factor .94 --gpus 0,1,2,3 --num-epoch 60 --network vgg`
Not converging: 0.001111, 0.000694, 0.000833, 0.001111, 0.000833, 0.000833, 0.000972, 0.000278

**Comment by hfutxrg** [comment]
I also got the same error in googlenet and other networks. Can anyone explain what's the values in kernel[0], kernel[1], kernel[2] and pad[0] and pad[1]? I tried to use gdb python but found I do not have it and could not install it right now. Any help now is appreciated.

**Comment by rongrongxiangxin** [comment]
@mli For resnet, what's the input size ? is 299 ok?

**Comment by hoaphumanoid** [comment]
Should be solved with #3506

**Comment by blazerygy** [comment]
how to solve the googlenet problem? @miguelgfierro @mli @piiswrong thx
